server/lib/chat/chat-api.ts

```diff
@@ -156,6 +168,27 @@ export class ChatApi {
     return await this.chatService.joinChannel(channelName, ctx.session!.user!.id)
   }
 
+  @httpPatch('/:channelId')
+  @httpBefore(
+    throttleMiddleware(editThrottle, ctx => String(ctx.session!.user!.id)),
+    handleMultipartFiles(MAX_IMAGE_SIZE),
+  )
+  async editChannel(ctx: RouterContext): Promise<EditChannelResponse> {
+    const channelId = getValidatedChannelId(ctx)
+    const {
+      body: { jsonBody },
+    } = validateRequest(ctx, {
+      body: Joi.object<{ jsonBody: EditChannelRequest }>({
```

Comment:
`jsonBody` could probably be named better for this request now that it's not a magic string